src/snap/plugin.cc

```diff
@@ -52,7 +53,46 @@ Plugin::Meta::Meta(Type type, std::string name, int version) :
                      concurrency_count(5),
                      exclusive(false),
                      cache_ttl(std::chrono::milliseconds(500)),
-                     strategy(Strategy::LRU) {}
+                     strategy(Strategy::LRU),
+                     tls_enabled(false),
+                     tls_certificate_key_path(""),
+                     tls_certificate_crt_path(""),
+                     tls_certificate_authority_path(""),
+                     tls_cipher_suites("") { 
+                       const char* tls_env = getenv("TLS_ENABLED");
+                       if (tls_env != nullptr && strcmp(tls_env, "false") != 0) {
+                         std::stringstream message_template{};
+                         message_template << "Encryption enabled but ";
+
+                         const char* taoCS = getenv("TAOCS");
+                         const char* taoCC = getenv("TAOCC");
+                         const char* taoCK = getenv("TAOCK");
+                         const char* taoCA = getenv("TAOCA");
+
```

Comment:
To enable TLS, example plugin should take flags from main arguments (argc, argv)- not from environment variables. This is implemented in latest closed pull request #46. 

https://github.com/intelsdi-x/snap-plugin-lib-cpp/blob/master/examples/collector/src/rando.cc#L208
```go
int main(int argc, char **argv) {
    Flags cli(argc, argv);
    Meta meta(Type::Collector, "rando", 1, &cli);

    if (cli.IsParsedFlag("version")) {
        cout << meta.name << " version "  << meta.version << endl;
        exit(0);
    }

    Rando plg;
    start_collector(&plg, meta);
}


```